functions/metadata.js

```diff
@@ -0,0 +1,74 @@
+// image (ENS logo but hue rotated by modulo of the hash)
+// block registered (?)
+import moment from 'moment';
+import addresses from './config/addresses';
+import web3Utils from 'web3-utils';
+import faunadb, {query as q} from 'faunadb';
+import {registrarContract, niftyContract} from './config/contracts';
+import {nodeFromLabelHash} from './helpers/hashes';
+import BigNumber from 'bignumber.js';
+
+require('dotenv').config();
+
+const client = new faunadb.Client({
+  secret: process.env.FAUNADB_SECRET,
+});
+
+function getOwner(labelHash) {
+  return niftyContract.methods
+    .ownerOf(labelHash)
+    .call()
+    .catch(() => `0x${'0'.repeat(40)}`);
+}
+
+function mod(numberOne, numberTwo) {
+  return BigNumber(`${numberOne}`)
+    .mod(BigNumber(`${numberTwo}`))
+    .toString();
+}
+
+async function getNameAttributes(labelHash) {
+  const owner = await getOwner(labelHash);
+  const entryInfo = await registrarContract.methods.entries(labelHash).call();
+  const dateRegistered = entryInfo[2];
+  const lockedEther = entryInfo[3];
+  return {
+    dateRegistered: moment(+dateRegistered * 1000).format('MM/DD/YYYY'),
+    lockedEther: web3Utils.fromWei(lockedEther),
+    owner,
+  };
+}
```

Comment:
maybe we should add this info when we register the name instead of when we serve the metadata. I know etherscan has a time limit for metadata calls and they won't accept the NFT if the response is too slow. just storing this stuff in the db would make the metadata endpoint much faster, no?